### PR TITLE
Update kernel source to 5.15.41.1 to address CVE-2022-28893, CVE-2022-29581

### PR DESCRIPTION
**Pull request by Camelron**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Update kernel source to 5.15.41.1 to address CVE-2022-28893, CVE-2022-29581. Also address old CVE-2020-35501, which we nopatched in 1.0 but forgot to also do in 2.0. 

###### Change Log  

- Update kernel source to 5.15.41.1
- Address CVE-2022-28893, CVE-2022-29581, CVE-2020-35501
- Make small changes to update_kernel.sh (don't auto-update kernel-rt, since we need to manually find the patch and it often is not ready yet, use the new mariner-2 tag naming scheme)

###### Does this affect the toolchain?  

YES

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-28893
- https://nvd.nist.gov/vuln/detail/CVE-2022-29581
- https://nvd.nist.gov/vuln/detail/CVE-2020-35501

###### Test Methodology

- amd: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=197717&view=results
- arm: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=197791&view=results

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -7192,7 +7192,7 @@
         "other": {
           "name": "kernel-rt",
           "version": "5.15.34.1",
-          "downloadUrl": "https://github.com/microsoft/CBL-Mariner-Linux-Kernel/archive/rolling-lts/mariner-2/5.15.34.1.tar.gz"
```
Comment (rlmenge) [inline cgmanifest.json:7195]: Was this an intended change?

File: SPECS/kernel/CVE-2022-29581.nopatch [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CVE-2022-29581 - patched in 5.15.36 - (generated by autopatch tool)
+upstream 3db09e762dc79584a69c10d74a6b98f89a9979f8 - stable ba9e9a794fd1689bf7e8a7452c55f3d3cbda7728
```
Comment (rlmenge) [inline SPECS/kernel/CVE-2022-29581.nopatch:2]: Nit: @nicogbg Could we add a new line in the nopatch after stable?

**Review by rlmenge — APPROVED** [review]
some nits but LGTM

File: cgmanifest.json (hunk 1 above)
Comment (Camelron) [reply cgmanifest.json:7195]: good catch!

**Pull request merged**